base/sparse/linalg.jl

```diff
@@ -941,3 +941,26 @@ end
 chol(A::SparseMatrixCSC) = error("Use cholfact() instead of chol() for sparse matrices.")
 lu(A::SparseMatrixCSC) = error("Use lufact() instead of lu() for sparse matrices.")
 eig(A::SparseMatrixCSC) = error("Use eigs() instead of eig() for sparse matrices.")
+
+function spcov(X::SparseMatrixCSC{Tv1,Ti1} where {Tv1,Ti1})
+    n,p = size(X)
+    means = mean(X, 1)
+
+    #Part 1 compute X'X
+    part1 = X'X
+
+    #Part 2 compute X'μ
+    sums = sum(X, 1)
+    part2 = sums'means
+
+    #Part 3 compute μ'μ
+    part3 = zeros(p,p)
+    for i in 1:p
```

Comment:
You might be able to get a marginal speed boost by adding an `@inbounds` annotation, which avoids bounds checking in `getindex` and the like. For example, this could be written as
```julia
@inbounds for i = 1:p, j = i:p
    part3[i,j] = part3[j,i] = means[i] * means[j] * n
end
```
The simultaneous assignment as I've done above may or may not help, I actually don't know.